### PR TITLE
Feature packages

### PR DESCRIPTION
**Pull request by TajammalCodeLab**

### CODE REVIEW

File: signInAndSignOut_feature/UI/PackagesViewController/Packages/PackageCell.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+extension PackageCell: bindingDataDelegate{
+    func didSetDelegates(_ delegate: PackagesDelegates, with indexpath: IndexPath?) {
+        self.delegate = delegate
+        self.indexPath = indexpath
+    }
+    
+    
+}
+extension UITableViewCell {
+    var parentViewController: UIViewController? {
```
Comment (ghost) [inline signInAndSignOut_feature/UI/PackagesViewController/Packages/PackageCell.swift]: What is the purpose of doing this ?

File: signInAndSignOut_feature/UI/PackagesViewController/Packages/PackageCell.swift [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        contentView.layer.masksToBounds = false
+        contentView.backgroundColor = .clear
+    }
+    
+
+    @IBAction func actionButtonTapped(_ sender: UIButton) {
+        showActionSheet()
+    }
+
+    private func showActionSheet() {
```
Comment (ghost) [inline signInAndSignOut_feature/UI/PackagesViewController/Packages/PackageCell.swift]: Need to trigger this from controller not here

File: signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        if editingStyle == .delete {
+            didDelete(for: indexPath) // Pass the entire IndexPath
+        }
+    }*/
+}
+
+extension PackagesViewController: PackagesDelegates{
+    
+    func didEdit(for indexpath: IndexPath?) {
+        let alert = UIAlertController(title: "Edit Title", message: "Enter new title", preferredStyle: .alert)
```
Comment (ghost) [inline signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift]: Why these strings are not called from StringConstants ?

File: signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        alert.addTextField { textField in
+            textField.text = self.packagesArr[indexpath!.row].title
+        }
+        let saveAction = UIAlertAction(title: "Save", style: .default) { _ in
+            if let newTitle = alert.textFields?.first?.text, !newTitle.isEmpty {
+                self.packagesArr[indexpath!.row].title = newTitle
+                self.tableViewPackages.reloadRows(at: [IndexPath(row: indexpath!.row, section: 0)], with: .automatic)
+            }
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
```
Comment (ghost) [inline signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift]: Call strings from String Constants

File: signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    }
+    // MARK: - IBAction -
+    
+    @IBAction func directTodashboardbtn(_ sender: Any) {
+        directTodashboard()
+    }
+    
+    // MARK: - Methods -
+    private func updateDataArr() {
+        tableViewPackages.register(UINib(nibName: "PackageCell", bundle: nil), forCellReuseIdentifier: "packagecell")
```
Comment (ghost) [inline signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift]: Still you're passing the same image on every package

Comment (ghost) [inline signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift]: Same mistake for String/Identifiers

File: signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+}
+// MARK: - Extensions-
+extension PackagesViewController: UITableViewDataSource, UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return packagesArr.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableViewPackages.dequeueReusableCell(withIdentifier: "packagecell", for: indexPath) as! PackageCell
```
Comment (ghost) [inline signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift]: Pass identifiers from Identifier Class

File: signInAndSignOut_feature/UI/PackagesViewController/Packages/PackageCell.swift [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        updateCellUI()
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        contentView.frame = contentView.frame.inset(by: UIEdgeInsets(top: 8, left: 16, bottom: 8, right: 16))
```
Comment (ghost) [inline signInAndSignOut_feature/UI/PackagesViewController/Packages/PackageCell.swift:23]: Do this  thing inside the didSet on the outlet not here

File: signInAndSignOut_feature/UI/PackagesViewController/Packages/PackageCell.swift [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    }
+    
+    
+    func updateData(for package: Packages) {
+            packageNameLabel.text = package.title
+        imageback.image = UIImage(named: package.image ?? "") // Assuming the Packages model has an image property
+        }
+    private func updateCellUI() {
+        containerView.layer.cornerRadius = 15
```
Comment (ghost) [inline signInAndSignOut_feature/UI/PackagesViewController/Packages/PackageCell.swift:33]: update the UI on didSet not here

**Review by ghost — CHANGES_REQUESTED** [review]

File: signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift (hunk 3 above)
Comment (TajammalCodeLab) [reply signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift]: Updated

File: signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift (hunk 4 above)
Comment (TajammalCodeLab) [reply signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift]: Updated

File: signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift (hunk 5 above)
Comment (TajammalCodeLab) [reply signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift]: updated

Comment (TajammalCodeLab) [reply signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift]: updated

File: signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift (hunk 6 above)
Comment (TajammalCodeLab) [reply signInAndSignOut_feature/UI/PackagesViewController/PackagesViewController.swift]: updated

**Comment by TajammalCodeLab** [comment]
resolved

**Pull request closed without merging**